src/components/Trade/Components/TradeQuotes/TradeQuote.tsx

```diff
@@ -49,35 +53,78 @@ const TradeQuoteLoading = () => {
 }
 
 type TradeQuoteLoadedProps = {
-  assetId: AssetId
   isActive?: boolean
   isBest?: boolean
-  quoteDifference?: string
-  protocol: string
+  quoteDifference: string
   protocolIcon?: string
-  gasFiatPrice: string
-  quoteAmountCryptoPrecision: string
-  onClick: (protocol: string) => void
+  swapperWithMetadata: SwapperWithQuoteMetadata
+  totalReceiveAmountCryptoPrecision: string | undefined
 }
 
 export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
-  assetId,
   isActive,
   isBest,
   quoteDifference,
-  protocol,
   protocolIcon,
-  gasFiatPrice,
-  quoteAmountCryptoPrecision,
-  onClick,
+  swapperWithMetadata,
+  totalReceiveAmountCryptoPrecision,
 }) => {
   const translate = useTranslate()
   const borderColor = useColorModeValue('blackAlpha.100', 'whiteAlpha.100')
   const greenColor = useColorModeValue('green.500', 'green.200')
   const hoverColor = useColorModeValue('blackAlpha.300', 'whiteAlpha.300')
   const focusColor = useColorModeValue('blackAlpha.400', 'whiteAlpha.400')
-  const asset = useAppSelector(state => selectAssetById(state, assetId))
-  return (
+
+  const { setTradeAmountsUsingExistingData } = useTradeAmounts()
+
+  const feeAssetFiatRate = useSwapperStore(state => state.feeAssetFiatRate)
+  const buyAsset = useSwapperStore(state => state.buyAsset)
+  const sellAsset = useSwapperStore(state => state.sellAsset)
+  const amount = useSwapperStore(state => state.amount)
+  const updateActiveSwapperWithMetadata = useSwapperStore(
+    state => state.updateActiveSwapperWithMetadata,
+  )
+  const action = useSwapperStore(state => state.action)
+
+  const handleSwapperSelection = (activeSwapperWithMetadata: SwapperWithQuoteMetadata) => {
+    updateActiveSwapperWithMetadata(activeSwapperWithMetadata)
+    setTradeAmountsUsingExistingData({ amount, action })
+  }
+
+  const { quote, inputOutputRatio } = swapperWithMetadata
+
+  const feeAsset = useAppSelector(state => selectFeeAssetByChainId(state, sellAsset?.chainId ?? ''))
+  if (!feeAsset)
+    throw new Error(`TradeQuoteLoaded: no fee asset found for chainId ${sellAsset?.chainId}!`)
+
+  const networkFeeFiat = feeAssetFiatRate
+    ? bnOrZero(fromBaseUnit(quote.feeData.networkFeeCryptoBaseUnit, feeAsset.precision))
+        .times(feeAssetFiatRate)
+        .toString()
+    : undefined
+
+  const protocol = swapperWithMetadata.swapper.name
+  const amountEntered = amount !== '0'
```

Comment:
Prefer `bnOrZero(amount).gt()` - we should never have negative values but might as well use heuristics that prevent it

Reply:
Resolved in https://github.com/shapeshift/web/pull/4079/commits/0543329e80fb6c0944b4790d45acc67db58a2064